src/draw/sw/lv_draw_sw.c

```diff
@@ -216,6 +216,32 @@ static int32_t lv_draw_sw_delete(lv_draw_unit_t * draw_unit)
 #endif
 }
 
+void lv_draw_sw_i1_to_argb8888(void ** buf_i1, void ** buf_argb8888, size_t buf_size_px)
+{
+    /*Extract the bits of I1 px_map and convert them to ARGB8888*/
+    uint32_t px_map_size = buf_size_px / (4 * 8) ;
+    uint32_t * argb_px_map_arith = *buf_argb8888;
+    uint8_t * buf_i1_arith = *buf_i1;
+    /*Skip the palette*/
+    buf_i1_arith += LV_COLOR_INDEXED_PALETTE_SIZE(LV_COLOR_FORMAT_I1) * 4;
+    for(uint32_t i = 0; i < px_map_size; i++) {
+        /*From MSB to LSB (pixel 0 to pixel 7 in a byte)*/
+        for(int bit = 7; bit >= 0; bit--) {
+            /*White*/
+            if(((*(buf_i1_arith + i) >> bit) & 1) == 1) {
+                *(argb_px_map_arith) = 0x00FFFFFF;
+            }
+            /*Black*/
+            else {
+                *(argb_px_map_arith) = 0x00000000;
+            }
+            argb_px_map_arith++;
+        }
+    }
+    /*Pass the pixel map handle*/
+    *buf_i1 = *buf_argb8888;
```

Comment:
Originally this function resided in sdl related files, therefore it was not accessable by the user, but since it has been moved to sw_draw, it makes sense to elliminate this side effect. Thanks for pointing it out!